scripts/Packages/Package_Gullwing__QFP_SOIC_SO/Readme.md

```diff
@@ -0,0 +1,100 @@
+# IPC gullwing generator
+
+This generator uses IPC-7351B equations and by default fillet definitions to derive a footprint for gullwing stile packages.
+Examples of supported packages include: QFN, SOIC, SO
+
+## Running the script
+
+The script requires python 3.4 or newer. Run it with:
+`python3 ipc_gullwing_generator.py size_definitions/eqfp.yaml` (replace eqfp.yaml with the size definition file that contains your part.)
+
+## Size definition format
+
+Every file contains a header with parameters applied to all parts. These define the common footprint name prefix and the output library.
+
+```yaml
+FileHeader:
+  library_Suffix: 'QFP' #resulting library name Package_QFP
+  device_type: 'EQFP' #footprint names will be of style ...EQFP-pincount_...
+```
+
+---
+
+Every further entry in the script is assumed to be a package size definition. The top level parameter will give the internal parameter set name. This one must be unique in this file and should be representative of the footprint as it will be used in error messages. (It also makes later maintenance easier if it is easy to determine which footprint was genererated by which parameter set.)
+
+``` yaml
+internal_name:
+  # parameter list
+```
+
+The following parameters are supported for every package
+- Size source to be added to footprint documentation field (`size_source`) {url}
```

Comment:
IMO it would be worth considering if the parameter name should go first and then the description; right now they're backwards

Reply:
Both have their merits. I prefer this style as it allows combining multiple parameters.

Reply:
Then perhaps mention it's possible use as a header for the entire YAML file?

Reply:
I am not sure how your suggestion fits with the source code line this is attached to.